src/udiskslinuxfilesystem.c

```diff
@@ -150,53 +150,20 @@ get_filesystem_size (UDisksLinuxBlockObject *object)
   guint64 size = 0;
   UDisksLinuxDevice *device;
   gchar *dev;
-  const gchar *type;
   GError *error = NULL;
 
   device = udisks_linux_block_object_get_device (object);
   dev = udisks_linux_block_object_get_device_file (object);
-  type = g_udev_device_get_property (device->udev_device, "ID_FS_TYPE");
 
-  if (g_strcmp0 (type, "ext2") == 0)
+  size = bd_fs_get_size (dev, &error);
```

Comment:
FYI, this return value is directly exposed further down the code. Fortunately `bd_fs_get_size()` return value of `0` on error is compatible but care must be taken when breaking the `libblockdev` API. Nothing serious though.